heat/core/operations.py

```diff
@@ -755,8 +758,22 @@ def __reduce_op(x, partial_op, op, axis, out):
                 partial[0].dtype), split=out.split, comm=x.comm)
             return out
         return tensor.tensor(partial, output_shape, types.canonical_heat_type(partial[0].dtype), split=None, comm=x.comm)
-    if out is not None:
-        out._tensor__array = tensor.tensor(partial, output_shape, types.canonical_heat_type(
-            partial[0].dtype), split=out.split, comm=x.comm)
-        return out
-    return tensor.tensor(partial, output_shape, types.canonical_heat_type(partial[0].dtype), split=None, comm=x.comm)
+    else:
+        if out is not None:
+            out._tensor__array = tensor.tensor(partial, output_shape, types.canonical_heat_type(
+                partial[0].dtype), split=out.split, comm=x.comm)
+            return out
+        return tensor.tensor(partial, output_shape, types.canonical_heat_type(partial[0].dtype), split=None, comm=x.comm)
+
+
+def __sanitize_out(out, out_shape):
```

Comment:
Solid implementation, should be part of stride_tricks.py however.

Reply:
Thanks, will do.